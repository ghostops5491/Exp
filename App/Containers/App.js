import 'App/Config'
import DebugConfig from 'App/Config/DebugConfig'
import React, { Component } from 'react'
import { Provider } from 'react-redux'
import { View, Platform, SafeAreaView } from 'react-native'
import { Text } from 'native-base'
import RootContainer from './RootContainer'
import store from 'App/Redux'
import Actions from 'App/Redux/Actions'
import PushService from 'App/Services/PushService'
import { Sentry } from 'react-native-sentry'
import { getResponsiveCSSFrom8 } from '../Lib/Utils'
import TestFairy from 'react-native-testfairy'

// Comment out in development
Sentry.config(
  'https://75b40415c7ee40268c3166a648081e53:dccca78cbe8c4efeab3e33a7e196d58f@sentry.io/1187460'
).install()

const pushService = new PushService(
  payload => store.dispatch(Actions.savePushTokenLocal(payload)),
  payload => store.dispatch(Actions.notificationReceived(payload))
)
pushService.configure()

class App extends Component {
  componentWillMount = () => {
    TestFairy.begin('1307d9f31b99123e4f1112e69856ecc9eb3c2258')
  }
  render() {
    return (
      <Provider store={store}>
        <SafeAreaView
          style={{
            flex: 1,
            backgroundColor: '#fcfcfc',
            ...Platform.select({
              ios: {
                marginTop: -20
              }
            })
          }}
        >
          <RootContainer />
          { DebugConfig.stagingBuild &&
            <Text style={{textAlign: 'center', color: '#ef6940'}}>STAGING BUILD</Text>
          }
        </SafeAreaView>
      </Provider>
    )
  }
}

export default App
