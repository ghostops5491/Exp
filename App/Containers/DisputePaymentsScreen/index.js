import React, {Component} from 'react'
import {View, Text, TouchableOpacity, Linking} from 'react-native'
import {getResponsiveCSSFrom8} from '../../Lib/Utils'
import {Content} from 'native-base'

const JOI_PAYMENTS_EMAIL = 'payments@jonnyonit.com'
export default class DisputePaymentsScreen extends Component {
  handleEmailPressed = () => {
    Linking.openURL(`mailto:${JOI_PAYMENTS_EMAIL}`)
  }

  render() {
    return (
      <View style={{
        flex: 1,
        padding: getResponsiveCSSFrom8(25).width,
        alignItems: 'center',
        justifyContent: 'center'
      }}>
        <View style={{
          alignSelf: 'stretch'
        }}>
          <Text style={{padding: 10, textAlign: 'center'}}>
            If you would like to dispute this payment, please email
          </Text>
          <TouchableOpacity
            style={{
              alignSelf: 'stretch'
            }}
            onPress={this.handleEmailPressed}>
            <Text style={{textAlign: 'center', color: 'blue'}}>
              {JOI_PAYMENTS_EMAIL}
            </Text>
          </TouchableOpacity>
          <Text style={{textAlign: 'center'}}>with your request.</Text>
        </View>
      </View>
    )
  }
}
