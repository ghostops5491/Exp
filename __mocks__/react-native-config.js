// __mocks__/react-native-config.js
export default {
    API_URL: 'http://localhost:3000/api/v1/',
  };