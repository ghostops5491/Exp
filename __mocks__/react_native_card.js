import React from 'react';

const constants = {
    DETECTION_MODE:{},
    CardIOUtilities:{}
};

class Card extends React.Component {

  static constants = constants
  render() {
    return null;
  }
}

Card.constants = constants;

export default Card;