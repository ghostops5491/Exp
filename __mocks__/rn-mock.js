module.exports = {
    Dimensions : {
        get : jest.fn( () => ( { width : 10, height : 10 } ) )
    },

    Platform : {
        OS : "ios",
        select : jest.fn()
    },

    StyleSheet : {
        create : jest.fn()
    }
};