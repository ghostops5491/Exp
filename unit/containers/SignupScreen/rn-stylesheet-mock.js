/**
 * StyleSheet Mock
 * ~~~~~~~~~~~~~~~
 * 
 * StyleSheet mock for SignupScreen
 */

/**
 * Stub for create method on StyleSheet export of react-native
 * @param {object} styles 
 */
function create( styles ) {

    return Object.keys( styles )
                 .reduce( ( reducedStyles, style ) => Object.assign(
                     reducedStyles,
                     { [style] : { marginBottom : 10 } }
                 ), {} );
}

/** Expose the mock */
module.exports = { create };