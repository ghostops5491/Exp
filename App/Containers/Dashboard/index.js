import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Image, View, ScrollView, TouchableOpacity } from 'react-native'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'
import { Button, Text, Content } from 'native-base'
import { get, map } from 'lodash'
import { Colors } from 'App/Themes'
import styles from './styles'
import { MainContainer, Container, HeaderNav,
  FooterNav, Section, SectionList } from 'App/Components'
import Actions from 'App/Redux/Actions'
import JobItem from './JobItem'

class Dashboard extends Component {
  static propTypes = {
    fetchJob: PropTypes.func.isRequired,
    allJobs: PropTypes.object.isRequired,
  }

  componentWillMount() {
    this.props.fetchJobs()
      .catch(console.log)
    this.props.fetchUpcomingJobs()
      .catch(console.log)
  }

  gotoJobCategories = () => {
    this.props.setServiceRequestType('normal')
    this.props.navigation.navigate("JobCategories")
  }

  goToJobDetail(job) {
    this.props.setCurrentJob(job.id)
    const nextRoute = job.status == 'waiting' ? 'FindingProvider' : 'JobTimeline'
    this.props.navigation.navigate(nextRoute)
  }

  render () {
    return (
      <View style={{
        backgroundColor: Colors.grayThree,
        flex: 1
      }}>
        <Section>
          <View style={styles.buttonDescription}>
            <Text style={styles.text}>
              Find a qualified service provider now!
            </Text>
          </View>
          <Button primary
            style={{ width: '100%', height: 46, marginTop: 12 }}
            onPress={this.gotoJobCategories}
          >
            <Text style={styles.buttonText}> BOOK JOB </Text>
          </Button>
        </Section>

        <ScrollView>
          <SectionList
            title="ACTIVE JOBS"
            data={this.props.orderedActiveJobs}
            renderRow={(jobId) => {
              let job = this.props.allJobs[jobId]
              return (
                <TouchableOpacity
                  onPress={() => {this.goToJobDetail(job)}}
                >
                  <JobItem job={job} />
                </TouchableOpacity>
              )
            }}
          />

          <SectionList
            title="UPCOMING JOBS TODAY"
            data={this.props.orderedUpcomingJobs}
            renderRow={(jobId) => {
              let job = this.props.allJobs[jobId]
              return (
                <TouchableOpacity
                  onPress={() => {this.goToJobDetail(job)}}
                >
                  <JobItem job={job} />
                </TouchableOpacity>
              )
            }}
          />

          <SectionList
            title="PAST JOBS"
            data={this.props.orderedPastJobs}
            renderRow={(jobId) => {
              let job = this.props.allJobs[jobId]
              return (
                <TouchableOpacity
                  onPress={() => {this.goToJobDetail(job)}}
                >
                  <JobItem job={job} />
                </TouchableOpacity>
              )
            }}
          />
        </ScrollView>
      </View>
    )
  }
}

const mapStateToProps = createStructuredSelector({
  allJobs: (state) => get(state, 'job.allJobs'),
  orderedActiveJobs: (state) => get(state, 'job.orderedActiveJobs'),
  orderedUpcomingJobs: (state) => get(state, 'job.orderedUpcomingJobs'),
  orderedPastJobs: (state) => get(state, 'job.orderedPastJobs'),
})

const mapDispatchToProps = (dispatch) => ({
  fetchJob: (payload) =>  new Promise((resolve, reject) =>
    dispatch(Actions.fetchJob(payload, resolve, reject))),
  fetchJobs: () => {
    return new Promise((resolve, reject) =>
      dispatch(Actions.fetchJobsRequest(resolve, reject)))
  },
  fetchUpcomingJobs: () => {
    return new Promise ((resolve, reject) =>
      dispatch(Actions.fetchUpcomingJobsRequest(resolve, reject)))
  },
  setCurrentJob: (id) => dispatch(Actions.setCurrentJob(id)),
  setServiceRequestType: (rType) => dispatch(Actions.setServiceRequestType(rType)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard)
