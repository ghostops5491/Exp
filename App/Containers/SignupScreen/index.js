import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, TouchableOpacity, KeyboardAvoidingView, ActivityIndicator } from 'react-native'
import { Button, Text, Form, Item, Input, Icon } from 'native-base'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'
import TermsConditionsModal from 'App/Components/Modals/TermsConditionsModal'
import { AuthScreenLayout, RenderIcon } from 'App/Components'
import { get, isEmpty } from 'lodash'
import Actions from 'App/Redux/Actions'
import { Colors } from 'App/Themes'
import styles from './styles'

class SignupScreen extends Component {
  static propTypes = {
    clearError: PropTypes.func.isRequired,
    addError: PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props)

    this.state = {
      termsConditionsModalVisible: false,
      hidePassword1: true,
      hidePassword2: true
    }
  }

  componentWillMount() {
    this.props.clearForm()
  }

  signup() {
    if (this.validateInputs()) {
      this.props.clearError()
      this.toggleTermsConditionsModal()
    }
  }

  toggleTermsConditionsModal() {
    this.setState({
      termsConditionsModalVisible: !this.state.termsConditionsModalVisible
    })
  }

  acceptTerms = () => {
    const { email, password, passwordConfirmation, registerPushToken, deviceToken } = this.props

    this.toggleTermsConditionsModal()
    this.props .signup({ email, password, passwordConfirmation })
    .then(() => {
      this.props.clearForm()
      this.props.navigation.navigate('ProfileTabs')
      registerPushToken(deviceToken)
    })
    .catch(console.log)
  }

  declineTerms = () => {
    this.toggleTermsConditionsModal()
    this.props.navigation.navigate('LoginScreen')
  }

  validateInputs() {
    return this.validEmail() && this.validPassword() && this.validPasswordConfirmation()
  }

  validEmail = () => {
    const { email, addError } = this.props
    const emailRegex =/^(([^$-\/:-?{-~!"^`\[\]@\\#]+([.\-_+][^$-\/:-?{-~!"^`\[\]@\\]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    if (isEmpty(email)) {
      addError({ input: 'email', message: 'Please input email.' })
      return false
    } else if (!emailRegex.test(email)) {
      addError({ input: 'email', message: 'Please input valid email.' })
      return false
    }
    return true
  }

  validPassword() {
    const { password, addError } = this.props
    if (isEmpty(password)) {
      addError({ input: 'password', message: 'Please input password.' })
      return false
    } else if (password.length < 6) {
      addError({ input: 'password', message: 'Too short password. Input at least 6 characters.' })
      return false
    }
    return true
  }

  validPasswordConfirmation() {
    const { password, passwordConfirmation, addError } = this.props
    if (isEmpty(passwordConfirmation)) {
      addError({ input: 'passwordConfirmation', message: 'Please match password.' })
      return false
    } else if (password !== passwordConfirmation) {
      addError({
        input: 'passwordConfirmation',
        message: 'Password confirmation does not match with password.'
      })
      return false
    }
    return true
  }

  togglePassword = (index) => {
    if(index === 1) {
      this.setState({hidePassword1: !this.state.hidePassword1})
    } else {
      this.setState({hidePassword2: !this.state.hidePassword2})
    }
  }

  render() {
    const { error } = this.props
    const emailInputStyle = [styles.input, (error.input === 'email' && { borderColor: Colors.paleRed })]
    const passwordInputStyle = [styles.input, error.input === 'password' && { borderColor: Colors.paleRed }]
    const passwordConfirmationInputStyle = [styles.input, error.input === 'passwordConfirmation' && { borderColor: Colors.paleRed }]

    return (
      <AuthScreenLayout>
        <KeyboardAvoidingView>
          <View style={styles.formWrapper}>
            <View style={styles.hasError}>
              <Text style={styles.errorText}>
                {error && error.message}
              </Text>
            </View>
            <View>
              <Text style={styles.infoMessage}>
                Currently only serving the Atlanta, GA, metro area
              </Text>
            </View>
            <Form>
              <Item style={styles.item}>
                <View style={styles.icon}>
                  <RenderIcon name='username' />
                </View>
                <Input
                  placeholder='Email'
                  style={emailInputStyle}
                  value={this.props.email}
                  keyboardType='email-address'
                  autoCapitalize='none'
                  onChangeText={(email) =>
                    this.props.changeEmail(email)}
                />
              </Item>
              <Item style={styles.item}>
                <View style={styles.icon}>
                  <RenderIcon name='password' />
                </View>
                <Input
                  placeholder='Password'
                  secureTextEntry={this.state.hidePassword1}
                  style={passwordInputStyle}
                  value={this.props.password}
                  onChangeText={(password) =>
                    this.props.changePassword(password)}
                />
                <Button style={styles.rightIcon} transparent primary onPress={() => this.togglePassword(1)}>
                  <Icon name={this.state.hidePassword1 ? 'eye' : 'eye-off'}/>
                </Button>
              </Item>
              <Item style={styles.item}>
                <View style={styles.icon}>
                  <RenderIcon name='password' />
                </View>
                <Input
                  placeholder='Password Confirmation'
                  secureTextEntry={this.state.hidePassword2}
                  style={passwordConfirmationInputStyle}
                  value={this.props.passwordConfirmation}
                  onChangeText={(passwordConfirmation) =>
                    this.props.changePasswordConfirmation(passwordConfirmation)
                  }
                />
                <Button style={styles.rightIcon} transparent primary onPress={() => this.togglePassword(2)}>
                  <Icon name={this.state.hidePassword2 ? 'eye' : 'eye-off'}/>
                </Button>
              </Item>
              <View style={styles.activityIndicator}>
                {this.props.signingUp &&
                  <ActivityIndicator
                    size="large"
                    color={Colors.turtleGreen}
                  />}
              </View>
              <Item style={styles.item}>
                <Button primary
                  style={styles.loginButton}
                  onPress={() => this.signup()}
                >
                  <Text style={styles.loginButtonText}> SIGN UP </Text>
                </Button>
              </Item>
            </Form>
            <View style={styles.signInLine}>
              <Text style={styles.signInText}>
                Already have account?
              </Text>
              <TouchableOpacity
                style={styles.button}
                onPress={() => this.props.navigation.navigate('LoginScreen')}
              >
                <Text style={styles.signInLink}>
                  Log In!
                </Text>
              </TouchableOpacity>
            </View>
          </View>
          <TermsConditionsModal
            visible={this.state.termsConditionsModalVisible}
            acceptTerms={this.acceptTerms}
            declineTerms={this.declineTerms}
            showAcceptDeclineButtons={true}
          />
        </KeyboardAvoidingView>
      </AuthScreenLayout>
    )
  }
}

const mapStateToProps = createStructuredSelector({
  error: (state) => get(state, 'auth.form.error'),
  email: (state) => get(state, 'auth.form.email'),
  password: (state) => get(state, 'auth.form.password'),
  passwordConfirmation: (state) => get(state, 'auth.form.passwordConfirmation'),
  deviceToken: (state) => get(state, 'notification.deviceToken'),
  signingUp: (state) => get(state, 'auth.signingUp'),
  paymentMethodExists: (state) => get(state, 'payments.paymentMethodExists'),
  profileComplete: (state) => get(state, 'auth.profileComplete')
})

const mapDispatchToProps = dispatch => ({
  registerPushToken: (payload)  =>
    dispatch(Actions.registerPushTokenRequest(payload)),
  clearError: (payload)  => dispatch(Actions.clearError(payload)),
  addError: (payload)  => dispatch(Actions.addError(payload)),
  signup: (payload)  => {
    return new Promise((resolve, reject) =>
      dispatch(Actions.signupRequest(payload, resolve, reject)))
  },
  changeEmail: (email) => dispatch(Actions.changeEmail(email)),
  changePassword: (password) => dispatch(Actions.changePassword(password)),
  changePasswordConfirmation: (passwordConfirmation) =>
    dispatch(Actions.changePasswordConfirmation(passwordConfirmation)),
  clearForm: (email) => dispatch(Actions.clearForm(email))
})

export default connect(mapStateToProps, mapDispatchToProps)(SignupScreen)
