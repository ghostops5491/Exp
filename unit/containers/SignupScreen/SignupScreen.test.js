/**
 * SignupScreen Unit Tests
 * ~~~~~~~~~~~~~~~~~~~~~~~~
 */


/** Disable console errors */
console.error = () => {}

/** Manual mocks */
const dispatch = jest.fn();

/** Remocking react-native to support custom case of SignupScreen test cases */
jest.mock( "react-native", () => Object.assign ( 
    require( "./../../../__mocks__/rn-mock" ),
    { StyleSheet : require( "./rn-stylesheet-mock" ) }
) );

/** Package dependencies */
import React from "react";
import { shallow } from "enzyme";
import componentToJson from "enzyme-to-json";

/** Component in test */
import { SignupScreen } from "../../../App/Containers/SignupScreen";
/** 
 * Component props
 * ~~~~~~~~~~~~~~~ 
 * We need to explicitly define the component props
 * as we will be unit testing the component which is
 * not connected to Redis. So no mapStateToProps and
 * no mapDispatchToProps.
 * */
const props = {
    /** Props */
    error : "",
    email : "",
    password : "",
    passwordConfirmation : "",
    deviceToken : "",
    signingUp : false,
    paymentMethodExists : false,
    profileComplete : false,
    
    /** Dispatch methods */
    registerPushToken : jest.fn( payload => true ),
    clearError : jest.fn( payload => true ),
    addError : jest.fn( payload => true ),
    /** For now its a promise that resolves but the implmentation can be changed */
    signup : jest.fn( payload => Promise.resolve() ),
    changeEmail : jest.fn( payload => true ),
    changePassword : jest.fn( payload => true ),
    changePasswordConfirmation : jest.fn( password => true ),
    clearForm : jest.fn( payload => true )
},

/** Helper to get fresh props for each tests */
getProps = () => Object.assign( {}, props );

/** Tests for component state on initialization */
describe( "@method constructor", () => {

    test( "component state should match snapshot until changed explicitly", () => {
        /** 
         * Shallow rendering helps to get around the actual mounting 
         * and let ourselves be concerned about component properties
         * and its attributes at a particular state.
         */
        const component = shallow ( <SignupScreen { ...getProps() }/> );

        /** 
         * Call to matchSnapshot will either create a new snapshot
         * or assert against an already existing snapshot
         */
        expect( component.state() ).toMatchSnapshot();
    } );
} );

describe( "@method componentWillMount", () => {

    test( "should clear the form on component mount", () => {

        /** Hold a reference to props */
        const props = getProps();

        /** Shallow mount component to trigger componentWillMount */
        const component = shallow ( <SignupScreen { ...props }/> );

        expect( props.clearForm ).toHaveBeenCalledTimes( 1 );
    } );
} );

describe( "@behaviour signup", () => {

    test( "should clear errors and toggle terms and condition modal if inputs are valid", () => {

        /** Necessary mock */
        SignupScreen.prototype.validateInputs = jest.fn( () => true );

        /** Hold props */
        const props = getProps();

        /** Shallow mount component to trigger componentWillMount */
        const component         = shallow ( <SignupScreen { ...props }/> ),
              
              /** Hold modal state before signup */
              modalBeforeSignup = component.state( "termsConditionsModalVisible" );

        /** Trigger signup */
        component.instance().signup();

        const didClearErrorGotCalled    = props.clearError.mock.calls.length > 0,
              modalToggled              = component.state( "termsConditionsModalVisible" ) !== modalBeforeSignup;

        expect( didClearErrorGotCalled && modalToggled ).toBeTruthy();
    } );

    test( "should do nothing in case inputs are no good", () => {

        /** Necessary mock */
        SignupScreen.prototype.validateInputs = jest.fn( () => false );

        /** Hold props */
        const props = getProps();

        /** Shallow mount component to trigger componentWillMount */
        const component         = shallow ( <SignupScreen { ...props }/> ),
              
              /** Hold modal state before signup */
              modalBeforeSignup = component.state( "termsConditionsModalVisible" );

        /** Trigger signup */
        component.instance().signup();

        const didClearErrorGotCalled    = props.clearError.mock.calls.length > 0,
              modalToggled              = component.state( "termsConditionsModalVisible" ) !== modalBeforeSignup;

        expect( didClearErrorGotCalled && modalToggled ).toBeFalsy();
    } );
} );

test( "ui should match snapshot if not altered", () => {

    const component = shallow ( <SignupScreen { ...getProps() }/> );

    expect( componentToJson( component ) ).toMatchSnapshot();
} );