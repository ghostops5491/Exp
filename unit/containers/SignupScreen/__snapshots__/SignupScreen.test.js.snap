// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@method constructor component state should match snapshot until changed explicitly 1`] = `
Object {
  "hidePassword1": true,
  "hidePassword2": true,
  "termsConditionsModalVisible": false,
}
`;

exports[`ui should match snapshot if not altered 1`] = `
<AuthScreenLayout>
  <Component>
    <Component
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Component
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        />
      </Component>
      <Component>
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          Currently only serving the Atlanta, GA, metro area
        </mockConstructor>
      </Component>
      <mockConstructor>
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <RenderIcon
              name="username"
            />
          </Component>
          <mockConstructor
            autoCapitalize="none"
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="Email"
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
                false,
              ]
            }
            value=""
          />
        </mockConstructor>
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <RenderIcon
              name="password"
            />
          </Component>
          <mockConstructor
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
                false,
              ]
            }
            value=""
          />
          <mockConstructor
            onPress={[Function]}
            primary={true}
            style={
              Object {
                "marginBottom": 10,
              }
            }
            transparent={true}
          >
            <mockConstructor
              name="eye"
            />
          </mockConstructor>
        </mockConstructor>
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <RenderIcon
              name="password"
            />
          </Component>
          <mockConstructor
            onChangeText={[Function]}
            placeholder="Password Confirmation"
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "marginBottom": 10,
                },
                false,
              ]
            }
            value=""
          />
          <mockConstructor
            onPress={[Function]}
            primary={true}
            style={
              Object {
                "marginBottom": 10,
              }
            }
            transparent={true}
          >
            <mockConstructor
              name="eye"
            />
          </mockConstructor>
        </mockConstructor>
        <Component
          style={
            Object {
              "marginBottom": 10,
            }
          }
        />
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <mockConstructor
            onPress={[Function]}
            primary={true}
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <mockConstructor
              style={
                Object {
                  "marginBottom": 10,
                }
              }
            >
               SIGN UP 
            </mockConstructor>
          </mockConstructor>
        </mockConstructor>
      </mockConstructor>
      <Component
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <mockConstructor
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          Already have account?
        </mockConstructor>
        <Component
          onPress={[Function]}
        >
          <mockConstructor
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Log In!
          </mockConstructor>
        </Component>
      </Component>
    </Component>
    <TermsConditionsModal
      acceptTerms={[Function]}
      declineTerms={[Function]}
      showAcceptDeclineButtons={true}
      visible={false}
    />
  </Component>
</AuthScreenLayout>
`;
